presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -179,7 +179,10 @@ private void waitForMinimumWorkers()
     {
         ListenableFuture<?> minimumWorkerFuture = clusterSizeMonitor.waitForMinimumWorkers();
         // when worker requirement is met, wait for query execution to finish construction and then start the execution
-        addSuccessCallback(minimumWorkerFuture, () -> addSuccessCallback(queryExecutionFuture, this::startExecution));
+        addSuccessCallback(minimumWorkerFuture, () -> {
+            stateMachine.endWaitingForResources();
```

Comment:
after thinking further it feels like instead of just ending waiting for resources, we should move the timer to transitionToDispatching. This will make sure we count the gap (T2 - T1) as part of dispatching and not been missed from the accounting.